dandi/dandiarchive
# Add meta.dandiset.identifier index

Fixes #96 

Add a mongo index on `meta.dandiset.identifier`. When I did not include `parentId` in the index, Mongo decided to use one of the `parentId` indices instead, which is not useful. I opted for `meta.dandiset.identifier` before `parentId` so the index is still useful if we ever have an endpoint that does not specify the `parentId`.

Test steps:
1. Wipe database
2. Start server and set up user
3. Verify that the index is present
4. Create a new dandiset
5. GET /dandi with `identifier:"000001"`
6. GET /dand with `identifier:"000002"`
7. Verify that the index was used twice (`db.folder.aggregate( [ { $indexStats: { } } ] )`)
